### PR TITLE
Improve agent plugin documentation with running examples and additional notes

### PR DESCRIPTION
**Pull request by Sovietaced**
I recently went through the process of developing an agent plugin by going through the docs. Previously the `developing_agents` documentation had a decent skeleton for the agent example but the `local_python` documentation had little to no accompanying task example. Its a big confusing because the agent is a stripped down big query agent that is not functional and then later the example task is just a real big query task. 

Anyhow, I've spent some time updating the documentation to use a completely running agent, task, and example file. Previous agent pseudo code has been commented out and the general workflow has additional comments but there is dummy code that does actually run locally in a manner that someone can start making changes and get feedback immediately to start development. 

```
pyflyte run example.py say_hello --name jason                                                                                                                                                                                                                      ▼ Py base 01:03:20 AM
Running Execution on local.
create called task_template=<FlyteLiteral(TaskTemplate) id { resource_type: TASK name: "armada_test_agent.example.say_hello" } type: "example" metadata { runtime { type: FLYTE_SDK version: "1.12.0" flavor: "python" } retries { } } interface { inputs { variables { key: "name" value { type { simple: STRING } description: "name" } } } outputs { variables { key: "o0" value { type { simple: STRING } description: "o0" } } } } custom { fields { key: "environment" value { string_value: "dev" } } } container { image: "cr.flyte.org/flyteorg/flytekit:py3.11-1.12.0" args: "pyflyte-execute" args: "--inputs" args: "/tmp/flyte-od3iwm33/raw/0991ba2310db3416e9ad85aba218c0d9/inputs.pb" args: "--output-prefix" args: "/tmp/flyte-od3iwm33/raw/0991ba2310db3416e9ad85aba218c0d9" args: "--raw-output-data-prefix" args: "/tmp/flyte-od3iwm33/raw/0991ba2310db3416e9ad85aba218c0d9/raw_output" args: "--checkpoint-path" args: "/tmp/flyte-od3iwm33/raw/0991ba2310db3416e9ad85aba218c0d9/checkpoint_output" args: "--prev-checkpoint" args: "/tmp/flyte-od3iwm33/raw/0991ba2310db3416e9ad85aba218c0d9/prev_checkpoint" args: "--resolver" args: "flytekit.core.python_auto_container.default_task_resolver" args: "--" args: "task-module" args: "armada_test_agent.example" args: "task-name" args: "say_hello" resources { } }>
get called resource_meta=ExampleMetadata(job_id='temp')
```

### CODE REVIEW

File: docs/flyte_agents/testing_agents_in_a_local_python_environment.md [hunk 1]
```diff
@@ -14,48 +14,121 @@ You can test agents locally without running the backend server.
...
+get called resource_meta=ExampleMetadata(job_id='temp')
 ```
 
-You can also run a BigQuery task in your Python interpreter to test the agent locally.
+If it doesn't appear that your agent is running you can debug what might be going on by passing the `-v` verbose flag to `pyflyte`.
+
+## Existing Flyte Agent Tasks
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_in_a_local_python_environment.md:118]:
```suggestion
## Existing Flyte agent tasks
```

File: docs/flyte_agents/testing_agents_in_a_local_python_environment.md [hunk 2]
```diff
@@ -14,48 +14,121 @@ You can test agents locally without running the backend server.
...
+
+You can also run a existing Flyte Agent tasks in your Python interpreter to test the agent locally.
```
Comment (neverett) [inline docs/flyte_agents/testing_agents_in_a_local_python_environment.md:120]:
```suggestion
You can also run existing Flyte agent tasks in your Python interpreter to test the agent locally.
```

**Review by neverett — DISMISSED** [review]
A few small suggestions, otherwise LGTM!

**Comment by Sovietaced** [comment]
> A few small suggestions, otherwise LGTM!

Thanks for the review!

**Review by neverett — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
is there a way to re-trigger CI? (besides pushing a new commit)

**Comment by neverett** [comment]
@Sovietaced @davidmirror-ops Bumping this -- can you merge `master` in to re-trigger the tests and ensure no merge conflicts? If the tests pass, I think we should be able to merge this PR.

**Pull request closed without merging**